FirebaseMessaging/Sources/Token/FIRMessagingTokenInfo.m

```diff
@@ -172,7 +172,21 @@ - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
   FIRMessagingAPNSInfo *rawAPNSInfo = [aDecoder decodeObjectOfClasses:classes
                                                                forKey:kFIRInstanceIDAPNSInfoKey];
   if (rawAPNSInfo && ![rawAPNSInfo isKindOfClass:[FIRMessagingAPNSInfo class]]) {
-    return nil;
+    // If the decoder fails to decode a FIRMessagingAPNSInfo, check if this was archived by a
+    // FirebaseMessaging 10.18.0 or earlier.
+    @try {
+      [NSKeyedUnarchiver setClass:[FIRMessagingAPNSInfo class]
+                     forClassName:@"FIRInstanceIDAPNSInfo"];
```

Comment:
Why is there a difference between class names here? `FIRInstanceIDAPNSInfo` vs. `FIRMessagingAPNSInfo`

Reply:
Because `FIRInstanceIDAPNSInfo` is how it was written out in 10.18.0 and earlier - probably to be compatible with the old InstanceID token.